### PR TITLE
New password utils (#3999)

### PR DESCRIPTION
**Pull request by Andrii-Yezenitskyi**
Created a new endpoint to check if the user has a password set.
Created a new endpoint to set a password if the user has no password set.

### CODE REVIEW

**Review by yourromchikkkkk — APPROVED** [review]

**Review by Pavl0-Gural — APPROVED** [review]

**Review by danylohlynsky — APPROVED** [review]

File: core/src/main/java/greencity/exception/handler/CustomExceptionHandler.java [hunk 1]
```diff
@@ -272,4 +272,20 @@ public final ResponseEntity<Object> handlePasswordsDoNotMatchesException(Passwor
...
+     * @param ex {@link UserAlreadyHasPasswordException} which should be
+     *           intercepted.
+     * @return {@link ResponseEntity} contains http status and body with exception
+     *         message.
+     */
+    @ExceptionHandler(UserAlreadyHasPasswordException.class)
+    public final ResponseEntity<Object> handleUserAlreadyHasPasswordException(
+        UserAlreadyHasPasswordException ex, WebRequest request) {
+        ExceptionResponse response = new ExceptionResponse(getErrorAttributes(request));
+        log.trace(ex.getMessage(), ex);
```
Comment (VadymMakitra) [inline core/src/main/java/greencity/exception/handler/CustomExceptionHandler.java:288]: Do we need to log it?

File: core/src/main/java/greencity/security/controller/OwnSecurityController.java [hunk 2]
```diff
@@ -220,4 +211,41 @@ public ResponseEntity<UserAdminRegistrationDto> managementRegisterUser(
...
+        @ApiResponse(code = 201, message = HttpStatuses.CREATED),
+        @ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST),
+        @ApiResponse(code = 401, message = HttpStatuses.UNAUTHORIZED)
+    })
+    @PostMapping("/set-password")
+    public ResponseEntity<Object> setPassword(@Valid @RequestBody SetPasswordDto dto,
+        @ApiIgnore @AuthenticationPrincipal Principal principal) {
+        String email = principal.getName();
+        service.setPassword(dto, email);
+        return ResponseEntity.status(HttpStatus.CREATED).build();
```
Comment (VadymMakitra) [inline core/src/main/java/greencity/security/controller/OwnSecurityController.java:249]: I think status ok will be better here

File: core/src/main/java/greencity/exception/handler/CustomExceptionHandler.java (hunk 1 above)
Comment (thesvider) [reply core/src/main/java/greencity/exception/handler/CustomExceptionHandler.java:288]: agree. please remove trace logs.

File: service-api/src/main/java/greencity/security/dto/ownsecurity/HasPasswordDto.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+public class HasPasswordDto {
+    private Boolean hasPassword;
```
Comment (thesvider) [inline service-api/src/main/java/greencity/security/dto/ownsecurity/HasPasswordDto.java:13]: why Boolean and not primitive? is null value possible?

File: service/src/main/java/greencity/security/service/OwnSecurityServiceImpl.java [hunk 4]
```diff
@@ -405,4 +399,32 @@ private LocalDateTime calculateExpirationDate(Integer expirationTimeInHours) {
...
+        return user.getOwnSecurity() != null;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public void setPassword(SetPasswordDto dto, String email) {
+        User user = userRepo.findByEmail(email)
+            .orElseThrow(() -> new WrongEmailException(ErrorMessage.USER_NOT_FOUND_BY_EMAIL));
+        if (Boolean.TRUE.equals(hasPassword(email))) {
```
Comment (thesvider) [inline service/src/main/java/greencity/security/service/OwnSecurityServiceImpl.java:418]: why not just if (hasPassword(email)) ? user.getOwnSecurity() != null; - never returns null

**Pull request merged**